modules/search-service-impl/src/main/java/org/opencastproject/search/impl/solr/SolrRequester.java

```diff
@@ -209,6 +213,18 @@ public MediaPackage getMediaPackage() {
           return null;
         }
 
+        @Override
+        public AccessControlList getAccessControlList() {
+          final List<AccessControlEntry> entries = Schema.getOcAcl(doc)
```

Comment:
I assume this works… no idea about the internal data structures. Could the result here be `null` and could you run into a `NullPointerException`?

Reply:
Doesn't look like it can return `null` ([the impl](https://github.com/LukasKalbertodt/opencast/blob/3c57decfa224a104e462e2428994452243f0adb6/modules/search-service-impl/src/main/java/org/opencastproject/search/impl/solr/Schema.java#L613-L624)).